### PR TITLE
[mobile_static] Get mono/tests running on mobile_static

### PR DESCRIPTION
**Pull request by alexanderkyte**
This PR gets the tests in `mono/tests` and `mini/tests` working for the mobile_static profile. This allows the jenkins runner to run to completion with the mobile_static profile.

Functionally, this blacklists some tests, alters the test runner to aot the tests, builds ilasm.exe and runs with an external mono, and pulls in the information in `mcs/build` to describe how mcs should be invoked by these tests.

### CODE REVIEW

**Comment by alexanderkyte** [comment]
build

File: mono/tests/Makefile.am [hunk 1]
```diff
@@ -65,9 +95,58 @@ STRESS_TESTS_SRC=	\
 	process-stress.cs	\
 	assembly-load-stress.cs
 
+BAS_TEST_MOBILE_STATIC_BLACKLIST= \
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: typo: ~~BAS_~~ BASE_

minor nit: maybe call it BASE_TEST_MOBILE_STATIC_NOT_SUPPORTED. Blacklist for me implies there's some sort of active filtering going on, but it's just a separate list that's used or not.

File: mono/tests/Makefile.am [hunk 2]
```diff
@@ -444,6 +476,15 @@ BASE_TEST_CS_SRC=		\
 	pinvoke_ppcd.cs	\
 	bug-29585.cs
 
+if INSTALL_MOBILE_STATIC
+BASE_TEST_CS_SRC= \
+	$(BASE_TEST_CS_SRC_UNIVERSAL)
+else
+BASE_TEST_CS_SRC= \
+	$(BASE_TEST_MOBILE_STATIC_BLACKLIST) \
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: now that I see it here, better call it `BASE_TEST_CS_SRC_MOBILE_STATIC_NOT_SUPPORTED` to make it consistent with the other variables which use the  BASE_TEST_CS_SRC_ prefix

File: mono/tests/Makefile.am [hunk 3]
```diff
@@ -631,7 +672,11 @@ AOT_DISABLED_TESTS=constraints-load.exe
 TEST_CSC_SRC=			\
 	vararg.cs
 
-TEST_IL_SRC=			\
+MOBILE_STATIC_TEST_IL_SRC_BLACKLIST=	\
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: same here, use TEST_IL_SRC_ prefix

File: mono/tests/Makefile.am [hunk 4]
```diff
@@ -716,14 +767,27 @@ TESTSI_IL=$(TEST_IL_SRC:.il=.exe)
 TESTBS=$(BENCHSRC:.cs=.exe)
 STRESS_TESTS=$(STRESS_TESTS_SRC:.cs=.exe)
 
+PREREQSI_IL_AOT=$(PREREQ_IL_SRC:.il=.exe.dylib)
+PREREQSI_CS_AOT=$(PREREQ_CS_SRC:.cs=.exe.dylib)
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: AOT_SUFFIX or whatever it's called instead of .dylib

File: mono/tests/Makefile.am [hunk 5]
```diff
@@ -716,14 +767,27 @@ TESTSI_IL=$(TEST_IL_SRC:.il=.exe)
...
+endif
+
 %.exe: %.cs TestDriver.dll
-	$(MCS) -r:System.dll -r:System.Xml.dll -r:System.Core.dll -r:TestDriver.dll -r:Mono.Posix.dll -out:$@ $<
+	$(MCS_CMD) -r:System.dll -r:System.Xml.dll -r:System.Core.dll -r:TestDriver.dll $(TEST_DRIVER_MOBILE_STATIC_BLACKLIST) -out:$@ $<
+
+%.exe.dylib: %.exe 
+	$(TEST_RUNTIME) $(AOT_BUILD_FLAGS) $<
+
+%.dll.dylib: %.dll
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: same here

File: mono/tests/Makefile.am [hunk 6]
```diff
@@ -716,14 +767,27 @@ TESTSI_IL=$(TEST_IL_SRC:.il=.exe)
...
+PREREQSI_CS_AOT=$(PREREQ_CS_SRC:.cs=.exe.dylib)
+
 EXTRA_DIST=test-driver test-runner.cs $(TEST_CS_SRC_DIST) $(TEST_IL_SRC) \
 	$(BENCHSRC) $(STRESS_TESTS_SRC) stress-runner.pl $(PREREQ_IL_SRC) $(PREREQ_CS_SRC)
 
 %.exe: %.il
 	$(ILASM) -out:$@ $<
 
+if INSTALL_MOBILE_STATIC
+	TEST_DRIVER_MOBILE_STATIC_BLACKLIST=-r:Mono.Posix.dll
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: The name of the variable looks unrelated, why is this a blacklist?

File: mono/tests/Makefile.am [hunk 7]
```diff
@@ -747,8 +811,23 @@ test_platform: test-eglib-remap
...
 endif
 
+AOT_EXTRA_LIBS = 	\
+bug-382986-lib.dll.dylib	\
+bug-324535-il.dll.dylib	\
+bug-36848-a.dll.dylib	\
+bug-81691-b.dll.dylib	\
+bug-327438.2.exe.dylib	\
+bug-81466-lib.dll.dylib	\
+constraints-load.exe.dylib
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: same AOT_SUFFIX remark here

File: mono/tests/Makefile.am [hunk 8]
```diff
@@ -32,21 +49,34 @@ CLASS=$(mcs_topdir)/class/lib/$(DEFAULT_PROFILE)
...
+MCS_NO_LIB = $(TEST_RUNTIME) $(mcs_topdir)/class/lib/build/mcs.exe -unsafe -debug \
 	-noconfig -nologo \
 	-nowarn:0162 -nowarn:0168 -nowarn:0219 -nowarn:0414 -nowarn:0618 \
 	-nowarn:0169 -nowarn:1690 -nowarn:0649 -nowarn:0612 -nowarn:3021 \
-	-nowarn:0197
+	-nowarn:0197 $(PROFILE_MCS_FLAGS)
+
+MCS_CMD = $(MCS_NO_LIB) -lib:$(CLASS)
 
-MCS = $(MCS_NO_LIB) -lib:$(CLASS)
```
Comment (akoeplinger) [inline mono/tests/Makefile.am]: why did you rename this, it doesn't seem necessary?

File: runtime/Makefile.am [hunk 9]
```diff
@@ -109,10 +109,17 @@ cur_dir_cmd = pwd
 PLATFORM_PATH_SEPARATOR = :
 endif
 
+if INSTALL_MOBILE_STATIC
+MOBILE_STATIC_FILTER=grep -v ilasm
```
Comment (akoeplinger) [inline runtime/Makefile.am]: maybe add a small comment on how this is filtering

File: mono/tests/Makefile.am (hunk 8 above)
Comment (akoeplinger) [inline mono/tests/Makefile.am]: Ah, I see: 17024ca26d3fc49672553397a59d09bc51b0ab07

**Comment by akoeplinger** [comment]
@alexanderkyte I did a very quick pass and didn't see any red flags, so a preliminary LGTM from me but if someone else could take a look too that wouldn't hurt 😄

File: mono/tests/Makefile.am (hunk 8 above)
Comment (alexanderkyte) [inline mono/tests/Makefile.am]: It collides with the definition of MCS in the profile configuration in /mcs otherwise

File: mcs/build/profiles/mobile_static.make [hunk 10]
```diff
@@ -56,3 +56,11 @@ endif
 
 ALWAYS_AOT = yes
 
+INVARIANT_AOT_OPTIONS=nimt-trampolines=900,ntrampolines=8000,nrgctx-fetch-trampolines=256
+
+ifndef MONO_DISABLE_GSHAREDVT
+INVARIANT_AOT_OPTIONS:=$(INVARIANT_AOT_OPTIONS),ngsharedvt-trampolines=2800
```
Comment (kumpera) [inline mcs/build/profiles/mobile_static.make]: Could you add some comments on how you picked those numbers and where do they show up?

Comment (kumpera) [inline mcs/build/profiles/mobile_static.make]: Another thing, we're supported to support gsharedvt on all FullAOT targets. Which ones are we missing?

File: mono/mini/Makefile.am.in [hunk 11]
```diff
@@ -48,9 +48,19 @@ RUNTIME_EXECUTABLE = $(if $(BOEHM),$(top_builddir)/mono/mini/mono-boehm,$(top_bu
...
 RUNTIME_AOTCHECK = MONO_PATH="$(CLASS)$(PLATFORM_PATH_SEPARATOR)." $(RUNTIME_EXECUTABLE)
 
-CSC = $(mcs_topdir)/class/lib/build/mcs.exe
-MCS = CSC_SDK_PATH_DISABLED= $(MINI_RUNTIME) $(CSC) -unsafe -nowarn:0162 -nologo -noconfig -r:$(CLASS)/mscorlib.dll -r:$(CLASS)/System.dll -r:$(CLASS)/System.Core.dll
-ILASM = $(MINI_RUNTIME) $(CLASS)/ilasm.exe
+MINI_CSC = $(mcs_topdir)/class/lib/build/mcs.exe
+MINI_MCS = CSC_SDK_PATH_DISABLED= $(MINI_RUNTIME) $(MINI_CSC) -unsafe -nowarn:0162 -nologo -noconfig -r:$(CLASS)/mscorlib.dll -r:$(CLASS)/System.dll -r:$(CLASS)/System.Core.dll
+
+if INSTALL_MOBILE_STATIC
+include $(mcs_topdir)/build/config.make
```
Comment (kumpera) [inline mono/mini/Makefile.am.in]: Why do we need to include all those mcs makefiles here?

File: mono/mini/Makefile.am.in [hunk 12]
```diff
@@ -486,12 +496,25 @@ test_sources = 			\
 	gc-test.cs \
 	gshared.cs
 
-regtests=basic.exe basic-float.exe basic-long.exe basic-calls.exe objects.exe arrays.exe basic-math.exe exceptions.exe iltests.exe devirtualization.exe generics.exe basic-simd.exe
+
+regtests_UNIVERSAL=basic.exe basic-float.exe basic-long.exe basic-calls.exe objects.exe arrays.exe basic-math.exe exceptions.exe iltests.exe devirtualization.exe generics.exe
+regtests_MOBILE_STATIC_BLACKLIST=basic-simd.exe
```
Comment (kumpera) [inline mono/mini/Makefile.am.in]: what's wrong with simd? Please leave a comment on why it was blacklisted.

File: mono/mini/Makefile.am.in [hunk 13]
```diff
@@ -48,9 +48,19 @@ RUNTIME_EXECUTABLE = $(if $(BOEHM),$(top_builddir)/mono/mini/mono-boehm,$(top_bu
...
+MINI_CSC = $(mcs_topdir)/class/lib/build/mcs.exe
+MINI_MCS = CSC_SDK_PATH_DISABLED= $(MINI_RUNTIME) $(MINI_CSC) -unsafe -nowarn:0162 -nologo -noconfig -r:$(CLASS)/mscorlib.dll -r:$(CLASS)/System.dll -r:$(CLASS)/System.Core.dll
+
+if INSTALL_MOBILE_STATIC
+include $(mcs_topdir)/build/config.make
+include $(mcs_topdir)/build/platforms/$(PLATFORM).make
+include $(mcs_topdir)/build/profiles/$(DEFAULT_PROFILE).make
+include $(mcs_topdir)/build/monolite_or_extern.make
+
+ILASM = $(with_mono_path) $(PROFILE_RUNTIME) $(CLASS)/ilasm.exe
```
Comment (kumpera) [inline mono/mini/Makefile.am.in]: To keep with the renaming above, should we prefix ILASM with MINI_ as well?

File: mono/mini/Makefile.am.in [hunk 14]
```diff
@@ -724,26 +747,40 @@ gsharedvtcheck:
 
 fullaot_regtests = $(regtests) aot-tests.exe $(if $(GSHAREDVT),gshared.exe)
 
-FULLAOT_LIBS = \
-	mscorlib.dll \
-	System.Core.dll \
-	System.dll \
+FULLAOT_LIBS_MOBILE_STATIC_BLACKLIST = \
```
Comment (kumpera) [inline mono/mini/Makefile.am.in]: Why are those libs being blacklisted? Please add it as a comment so we can act on it over time.

File: mono/tests/Makefile.am [hunk 15]
```diff
@@ -1,7 +1,24 @@
+
+# This allows us to include makefiles from mcs/
+# they use topdir, not mcs_topdir and expect PROFILE
+# to be set
+topdir = $(mcs_topdir)
+PROFILE = $(DEFAULT_PROFILE)
+
+include $(mcs_topdir)/build/config.make
```
Comment (kumpera) [inline mono/tests/Makefile.am]: Why do we need to include the mcs headers here?

File: mono/tests/Makefile.am [hunk 16]
```diff
@@ -65,9 +95,58 @@ STRESS_TESTS_SRC=	\
 	process-stress.cs	\
 	assembly-load-stress.cs
 
+BASE_TEST_MOBILE_STATIC_NOT_SUPPORTED= \
```
Comment (kumpera) [inline mono/tests/Makefile.am]: I lot of those tests don't look to include things not suppported by FullAOT.
Are some on this list because they are just broken but supported to work?

File: mono/tests/Makefile.am [hunk 17]
```diff
@@ -632,7 +673,11 @@ AOT_DISABLED_TESTS=constraints-load.exe
 TEST_CSC_SRC=			\
 	vararg.cs
 
-TEST_IL_SRC=			\
+IL_SRC_MOBILE_STATIC_NOT_SUPPORTED=	\
+	constraints-load.il \
```
Comment (kumpera) [inline mono/tests/Makefile.am]: There's no reason on why constraints-load.il is not supported.
My point is that there's a distinction between not supported and broken. We must make sure we don't confuse the two.

File: mono/mini/Makefile.am.in (hunk 14 above)
Comment (alexanderkyte) [inline mono/mini/Makefile.am.in]: These are libs that we don't compile in the bcl on the mobile_static profile. They're not compiled on mobile platforms.

**Comment by kumpera** [comment]
LGTM, we can address moving config up later. @marek-safar are you ok with this plan?

**Comment by alexanderkyte** [comment]
I'm running into a few symbols not being fully propagated from a clean clone. Let me fix up one or two things.

**Comment by marek-safar** [comment]
yes but rebase it on master first

**Comment by alexanderkyte** [comment]
Oh yeah, github is doing the github thing.

File: mono/tests/Makefile.am (hunk 17 above)
Comment (alexanderkyte) [inline mono/tests/Makefile.am]: On AOT, we will try to resolve the type BrokenIComparable from mscorlib at AOT time, which will fail. This test is disabled because it cannot be AOTed.

File: mono/tests/Makefile.am (hunk 16 above)
Comment (alexanderkyte) [inline mono/tests/Makefile.am]: They're disabled because they attempted to use an appdomain feature or because they emit remoting wrappers that we don't support. I'll go through and confirm that this is true for all of them though.

**Comment by kumpera** [comment]
@monojenkins merge

**Comment by kumpera** [comment]
LGTM

**Comment by alexanderkyte** [comment]
The only failing tests are as follows:

```

=============== EOF ===============
=============== finalizer-wait.exe.stderr ===============

Unhandled Exception:
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'handle'.
  at System.Runtime.InteropServices.SafeHandle.DangerousAddRef (System.Boolean& success) [0x00025] in <filename unknown>:0 
  at System.Threading.NativeEventCalls.SetEvent (Microsoft.Win32.SafeHandles.SafeWaitHandle handle) [0x00002] in <filename unknown>:0 
  at System.Threading.EventWaitHandle.Set () [0x00018] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.Threading.EventWaitHandle:Set ()
  at P.Finalize () [0x00001] in <filename unknown>:0 
[MVID] 2b06b12ce62b46e58613c4314b2aed5c 0,1,2
[MVID] fb3e78e11d4a4b77b0a9e6e0572c7933 4

=============== EOF ===============
make[3]: Leaving directory `/mnt/jenkins/workspace/test-mono-pull-request-amd64/mono/tests'
make[2]: Leaving directory `/mnt/jenkins/workspace/test-mono-pull-request-amd64/mono/tests'
```

and

```

Test Case Failures:
1) MonoTests.System.Net.Sockets.SocketTest.SendAsyncFile : System.Exception : Could not abort registered blocking threads before closing socket.
Thread StackTrace:
  at System.Net.Sockets.SafeSocketHandle.RegisterForBlockingSyscall () [0x00057] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/SafeSocketHandle.cs:114 
  at System.Net.Sockets.Socket.SendFile_internal (System.Net.Sockets.SafeSocketHandle safeHandle, System.String filename, System.Byte[] pre_buffer, System.Byte[] post_buffer, System.Net.Sockets.TransmitFileOptions flags) [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/Socket.cs:2944 
  at System.Net.Sockets.Socket.SendFile (System.String fileName, System.Byte[] preBuffer, System.Byte[] postBuffer, System.Net.Sockets.TransmitFileOptions flags) [0x00028] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/Socket.cs:2893 
  at System.Runtime.Remoting.Messaging.AsyncResult.Invoke (System.Runtime.Remoting.Messaging.AsyncResult ) [0x00000] in <filename unknown>:0 
  at System.Runtime.Remoting.Messaging.AsyncResult.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/corlib/System.Runtime.Remoting.Messaging/AsyncResult.cs:213 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00096] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:855 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1210 
[MVID] 2b06b12ce62b46e58613c4314b2aed5c 3,4,5,6
[MVID] 59fee0d3754549a8b883d76e39b49bf9 0,1,2


at System.Net.Sockets.SafeSocketHandle.ReleaseHandle () [0x000c6] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/SafeSocketHandle.cs:67
at System.Runtime.InteropServices.SafeHandle.DangerousReleaseInternal (System.Boolean dispose) [0x000d2] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:227
at System.Runtime.InteropServices.SafeHandle.InternalDispose () [0x00011] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/corlib/System.Runtime.InteropServices/SafeHandle.cs:156
at System.Runtime.InteropServices.SafeHandle.Dispose (System.Boolean disposing) [0x00006] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:262
at System.Runtime.InteropServices.SafeHandle.Dispose () [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/referencesource/mscorlib/system/runtime/interopservices/safehandle.cs:254
at System.Net.Sockets.Socket.Dispose (System.Boolean disposing) [0x00047] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/Socket.cs:3322
at System.Net.Sockets.Socket.Dispose () [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/Socket.cs:3328
at System.Net.Sockets.Socket.Close () [0x00007] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/Socket.cs:3257
at MonoTests.System.Net.Sockets.SocketTest.SendAsyncFile () [0x0013b] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/Test/System.Net.Sockets/SocketTest.cs:4282
at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00038] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/corlib/System.Reflection/MonoMethod.cs:309
[MVID] 06d5dbfa254349bca23fdaa72b8ce57e 8
[MVID] 2b06b12ce62b46e58613c4314b2aed5c 1,2,3,4,10
[MVID] 59fee0d3754549a8b883d76e39b49bf9 0,5,6,7

2) MonoTests.System.Net.WebRequestTest.TestReceiveCancelation : System.IO.IOException : Unable to write data to the transport connection: Connection reset by peer.
  ----> System.Net.Sockets.SocketException : Connection reset by peer
at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x000e8] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/referencesource/System/net/System/Net/Sockets/NetworkStream.cs:602
at System.Net.ResponseStream.InternalWrite (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00029] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net/ResponseStream.cs:132
at System.Net.ResponseStream.Close () [0x000a1] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net/ResponseStream.cs:95
at System.Net.HttpConnection.Close (System.Boolean force_close) [0x00018] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net/HttpConnection.cs:443
at System.Net.HttpListener.Cleanup (System.Boolean close_existing) [0x0004e] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net/HttpListener.cs:233
at System.Net.HttpListener.Close (System.Boolean force) [0x0000c] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net/HttpListener.cs:220
at System.Net.HttpListener.Close () [0x0001f] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net/HttpListener.cs:212
at MonoTests.System.Net.WebRequestTest.TestReceiveCancelation () [0x00173] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/Test/System.Net/WebRequestTest.cs:459
at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00038] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/corlib/System.Reflection/MonoMethod.cs:309
[MVID] 06d5dbfa254349bca23fdaa72b8ce57e 7
[MVID] 2b06b12ce62b46e58613c4314b2aed5c 9
[MVID] 59fee0d3754549a8b883d76e39b49bf9 0,1,2,3,4,5,6
--SocketException
at System.Net.Sockets.Socket.Send (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.Net.Sockets.SocketFlags flags) [0x00031] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/System/System.Net.Sockets/Socket.cs:2385
at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x000b4] in /mnt/jenkins/workspace/test-mono-pull-request-amd64/mcs/class/referencesource/System/net/System/Net/Sockets/NetworkStream.cs:591
[MVID] 59fee0d3754549a8b883d76e39b49bf9 0,1
```

I do not think that either are caused by my changes.

**Comment by akoeplinger** [comment]
@alexanderkyte yeah, those are known to be flaky right now (afaik Ludo looks in the finalizer-wait.exe and Marcos into Socket stuff).
